### ISSUE TITLE
debugging pingresp not received, disconnecting error

### ISSUE THREAD

**Issue by Ashtonian** [eclipse/paho.mqtt.golang#210]
I have a go client connecting to the latest mosquitto server and I'm getting disconnect issues. 
```
4/17/2018 7:16:46 PM[MQTT-CRIT] 00:16:46 ping.go:60: [pinger]   pingresp not received, disconnecting
4/17/2018 7:16:46 PM[MQTT-ERRO] 00:16:46 net.go:429: [net]      error triggered, stopping
4/17/2018 7:16:46 PM[MQTT-WARN] 00:16:46 net.go:150: [net]      incoming dropped a received message during shutdown
4/17/2018 7:16:46 PM[MQTT-WARN] 00:16:46 net.go:416: [net]      logic stopped
4/17/2018 7:16:46 PM00:16:46 run.go:32: mqtt2nsq-[ERROR]: Disconnect from mqtt error:pingresp not received, disconnecting
```

I can see on the server when I enable debug logs the server gets the pingreq and immediately responds with pingresp but the client obviously never gets it. Trying to figure out how to find the issue, I think it may be getting blocked by a security group or something in aws but then wouldn't the whole tcp connection not work? Its running in a docker image on rancher in ec2. The broker is in a different region currently but its public and open as far as routing is concerned. 

Any thoughts/ideas on where I can look ?

**Comment by alsm** [comment]
I'm afraid I don't have any experience with running mitt stuff in AWS, I hoped that someone else familiar with it might have been able to help. Have you been able to run your configuration locally to verify it works correctly outside AWS?

**Comment by Ashtonian** [comment]
We figured it out. The driver was getting overloaded and was not able to respond to the ping before the timeout period. With `Opts.SetOrderMatters(false)` the driver was able to keep up however the memory usage on the application was out of control and we didn't see a way to limit the number of goroutines created as the result of receiving a packet. We ended up forking it and creating an in memory queue rather then spin up a new goroutine every packet. 

Just for some additional feedback - It would be nice if there was some visibility into the load the driver is under, also wasn't very clear that by default the driver is not concurrent.
